lms/views/dashboard/api/course.py

```diff
@@ -7,13 +11,71 @@
     APICourse,
     APICourses,
     CourseMetrics,
+    Pagination,
 )
-from lms.models import RoleScope, RoleType
+from lms.models import Course, RoleScope, RoleType
 from lms.security import Permissions
 from lms.services.h_api import HAPI
 from lms.services.organization import OrganizationService
+from lms.validation._base import PyramidRequestSchema
 from lms.views.dashboard.base import get_request_course, get_request_organization
 
+PAGINATION_LIMIT = 100
+"""Maximum number of items to return in paginated endpoints"""
+
+
+def build_courses_pagination(
+    request, courses, limit: int = PAGINATION_LIMIT
+) -> tuple[list[Course], Pagination]:
+    """Build the pagination details for a list of courses.
```

Comment:
It wasn't immediately clear here that `list[Course]` is a single page of results. A docstring like "Return the first page and pagination metadata from a list of courses" would make that clear.